contracts/Lens/JoeLensV2.sol

```diff
@@ -0,0 +1,338 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.5.16;
+pragma experimental ABIEncoderV2;
+
+import "../JErc20.sol";
+import "../Joetroller.sol";
+import "../JToken.sol";
+import "../PriceOracle/PriceOracle.sol";
+import "../EIP20Interface.sol";
+import "../Exponential.sol";
+import "../IRewardDistributor.sol";
+
+/**
+ * @notice This is a version of JoeLens that contains write transactions
+ * and pulls reward speeds from a RewardDistributor. JoeLensV2 mainly makes calls 
+ * to various contracts to retrieve market/account data from our lending platform  
+ * and wraps it up nicely to be sent to the frontend.
+ * @dev Call these functions as dry-run transactions for the frontend.
+ */
+contract JoeLensV2 is Exponential {
+    /** 
+     * @notice Administrator for this contract 
+     */
+    address public admin;
+
+    /** 
+     *@notice The module that handles reward distribution 
+     */
+    address payable public rewardDistributor;
+
+    /** 
+     * @notice The native token symbol for this contract 
+     */
+    string public nativeSymbol;
+
+    struct JTokenMetadata {
+        address jToken;
+        uint256 exchangeRateCurrent;
+        uint256 supplyRatePerSecond;
+        uint256 borrowRatePerSecond;
+        uint256 reserveFactorMantissa;
+        uint256 totalBorrows;
+        uint256 totalReserves;
+        uint256 totalSupply;
+        uint256 totalCash;
+        uint256 totalCollateralTokens;
+        bool isListed;
+        uint256 collateralFactorMantissa;
+        address underlyingAssetAddress;
+        uint256 jTokenDecimals;
+        uint256 underlyingDecimals;
+        JoetrollerV1Storage.Version version;
+        uint256 collateralCap;
+        uint256 underlyingPrice;
+        bool supplyPaused;
+        bool borrowPaused;
+        uint256 supplyCap;
+        uint256 borrowCap;
+        uint256 supplyJoeRewardsPerSecond;
+        uint256 borrowJoeRewardsPerSecond;
+        uint256 supplyAvaxRewardsPerSecond;
+        uint256 borrowAvaxRewardsPerSecond;
+    }
+
+    struct JTokenBalances {
+        address jToken;
+        uint256 jTokenBalance;
+        uint256 balanceOfUnderlyingCurrent;
+        uint256 supplyValueUSD;
+        uint256 collateralValueUSD;
+        uint256 borrowBalanceCurrent;
+        uint256 borrowValueUSD;
+        uint256 underlyingTokenBalance;
+        uint256 underlyingTokenAllowance;
+        bool collateralEnabled;
+    }
+
+    struct AccountLimits {
+        JToken[] markets;
+        uint256 liquidity;
+        uint256 shortfall;
+        uint256 totalCollateralValueUSD;
+        uint256 totalBorrowValueUSD;
+        uint256 healthFactor;
+    }
+
+    /**
+     * @notice Constructor function that initializes the native symbol and administrator for this contract
+     * @param _nativeSymbol The native symbol of the chain, such as jAVAX
```

Comment:
This comment is not entirely correct. AVAX is the native symbol of the of the chain. jAVAX is the symbol of the Banker Joe market for the native gas token.

Reply:
I see, would you say that the variable naming would more accurately be like `_nativeSymbolMarket` then? And I'll change this comment to make it correct

Reply:
Yes that would be more precise, let's change.